src/commands/GameIntegration/overwatch.ts

```diff
@@ -0,0 +1,171 @@
+		player: string,
+		platform: PlatformUnion
+	) {
+		const EMBED_DATA = message.language.tget('COMMMAND_OVERWATCH_EMBED_DATA');
+		const ratings = this.ratingsToMap(overwatchData.ratings, r => r.role, r => r);
+
+		return new UserRichDisplay(
+			new MessageEmbed()
+				.setColor(getColor(message))
+				.setAuthor(EMBED_DATA.AUTHOR(overwatchData.name), this.kAuthorThumbnail)
+				.setTitle(EMBED_DATA.TITLE)
+				.setURL(`https://overwatchtracker.com/profile/${platform}/global/${player}`)
+				.setThumbnail(overwatchData.icon)
+		)
+			.addPage((embed: MessageEmbed) => embed
+				.setDescription([
+					EMBED_DATA.HEADERS.ACCOUNT,
+					EMBED_DATA.PLAYER_LEVEL(overwatchData.level),
+					EMBED_DATA.PRESTIGE_LEVEL(overwatchData.level + (overwatchData.prestige * 100)),
+					EMBED_DATA.TOTAL_GAMES_WON(overwatchData.gamesWon)
+				].join('\n'))
+				.addField(EMBED_DATA.RATINGS_TITLE, EMBED_DATA.RATINGS([
+					{ role: ratings.get('damage')!.role, level: ratings.get('damage')!.level },
+					{ role: ratings.get('support')!.role, level: ratings.get('support')!.level },
+					{ role: ratings.get('tank')!.role, level: ratings.get('tank')!.level },
+					{ role: 'average', level: overwatchData.rating }
+				])))
+			.addPage((embed: MessageEmbed) => embed.setDescription(this.extractStats(overwatchData, 'quickPlayStats', EMBED_DATA)))
+			.addPage((embed: MessageEmbed) => embed.setDescription(this.extractStats(overwatchData, 'competitiveStats', EMBED_DATA)))
+			.addPage((embed: MessageEmbed) => embed.setDescription(this.extractTopHeroes(overwatchData, 'quickPlayStats', EMBED_DATA)))
+			.addPage((embed: MessageEmbed) => embed.setDescription(this.extractTopHeroes(overwatchData, 'competitiveStats', EMBED_DATA)));
+	}
+
+	/**
+	 * Creates a `Map` using the `keyExtractor` and `valueExtractor` to obtain the key and value of each element
+	 * in an `Array` of `object`s
+	 * @remark If multiple elements have the same `key` then only the last element with that `key` will be included
+	 * @param inputArray The array of objects to transform into a Map
+	 * @param keyExtractor A function that describes where to find the `key` for the `Map`
+	 * @param valueExtractor A function that describes where to find the `value` for the `Map`
+	 * @returns a `Map<Key, Value>` of the values, mapped by the given key
+	 */
+	private ratingsToMap<I, K, V>(inputArray: I[], keyExtractor: (_: I) => K, valueExtractor: (_: I) => V): Map<K, V> {
+		return inputArray.reduce<Map<K, V>>((accumulator: Map<K, V>, element: I) => accumulator.set(keyExtractor(element), valueExtractor(element)), new Map<K, V>());
+	}
+
+	/** Retrieves the top 5 heroes (name and time played in milliseconds) for either `competitiveStats` or `quickPlayStats` */
+	private getTopHeroes(overwatchData: OverwatchDataSet, type: 'competitiveStats' | 'quickPlayStats'): TopHero[] {
+		return Object.keys(overwatchData[type].topHeroes)
+			.map(hero => {
+				if (Reflect.has(overwatchData[type].topHeroes, hero)) {
```

Comment:
Is this even possible to be false? You're mapping `Object.keys` of `overwatchData[type].topHeroes`, then checking if such property exists on it. *It always does*.